website/client-old/js/controllers/hallCtrl.js

```diff
@@ -1,16 +1,45 @@
 "use strict";
 
-habitrpg.controller("HallHeroesCtrl", ['$scope', '$rootScope', 'User', 'Notification', 'ApiUrl', 'Hall',
-  function($scope, $rootScope, User, Notification, ApiUrl, Hall) {
+habitrpg.controller("HallHeroesCtrl", ['$scope', '$rootScope', 'User', 'Notification', 'ApiUrl', 'Hall', 'Content',
+  function($scope, $rootScope, User, Notification, ApiUrl, Hall, Content) {
     $scope.hero = undefined;
     $scope.currentHeroIndex = undefined;
     $scope.heroes = [];
+    $scope.allItemPaths = getAllItemPaths();
 
     Hall.getHeroes()
       .then(function (response) {
         $scope.heroes = response.data.data;
       });
 
+    function _getFormattedItemReference (pathPrefix, keys, values) {
+      var finishedString = '\n'.concat('path: ', pathPrefix, ', ', 'value: {', values, '}\n');
+
+      _.each(keys, function (key) {
+        finishedString = finishedString.concat('\t', pathPrefix, '.', key, '\n');
+      });
+      return finishedString;
+    }
+
+    function getAllItemPaths () {
+      var quests = _getFormattedItemReference('items.quests', _.keys(Content.quests), 'Numeric Quantity');
```

Comment:
Heads up, we're migrating to lodash v4 (https://github.com/HabitRPG/habitica/pull/8495) which includes many breaking changes (https://github.com/lodash/lodash/wiki/Changelog#v400). Before merging this PR should be checked and updated to make sure all lodash methods are compatible with v4.